packages/keystone/databaseAdapters/BalancingReplicaKnexAdapter/adapter.js

```
@@ -13,179 +13,260 @@
+    );
+}
 
 class BalancingReplicaKnexAdapter extends KnexAdapter {
-    constructor ({ databaseUrl, replicaPools, routingRules }) {
-        super()
-        this._dbConnections = getNamedDBs(databaseUrl || conf['DATABASE_URL'])
-        const availableDatabases = Object.keys(this._dbConnections)
-        this._replicaPoolsConfig = getReplicaPoolsConfig(replicaPools || conf['DATABASE_POOLS'], availableDatabases)
-        this._routingRules = getQueryRoutingRules(routingRules || conf['DATABASE_ROUTING_RULES'], this._replicaPoolsConfig)
-    }
+    constructor({ databaseUrl, replicaPools, routingRules } = {}) {
```

ESLint

[error] Missing space before function parentheses.

(@typescript-eslint/space-before-function-paren)